### PR TITLE
Native support for ARM

### PR DESCRIPTION
**Pull request by shuangjj**
closes https://github.com/eris-ltd/eris-cli/issues/718

### CODE REVIEW

**Comment by zramsay** [comment]
LGTM, @benjaminbollen @silasdavis can we get a :+1: for the changes in the edb binary

File: erisdb/pipe/blockchain.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// +build !arm
```
Comment (benjaminbollen) [inline erisdb/pipe/blockchain.go]: what does this line do?

File: erisdb/pipe/blockchain_arm.go [hunk 2]
```diff
@@ -0,0 +1,269 @@
+package pipe
```
Comment (benjaminbollen) [inline erisdb/pipe/blockchain_arm.go]: what is different in blockchain_arm.go?

**Comment by benjaminbollen** [comment]
As an additional point, e-db is undergoing a restructuring, so we will have to port these changes to the new branches (ideally discuss-eris-gateway), but we can take this onboard as we make the transition

**Comment by benjaminbollen** [comment]
The changes in blockchain.go refer to `MaxUInt32` which is replaced by `MaxInt32`?  This only occurs in this file?

File: erisdb/pipe/blockchain_arm.go [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+	match func(int, int) bool
+}
+
+func (this *BlockHeightFilter) Configure(fd *FilterData) error {
+	op := fd.Op
+	var val int
+	if fd.Value == "min" {
+		val = 0
+	} else if fd.Value == "max" {
+		val = math.MaxInt32
```
Comment (shuangjj) [inline erisdb/pipe/blockchain_arm.go]: `var val int` should be signed 32-bit integer, but the assignment `val = math.MaxUint32` will overflow it. I tried to change the val type to uint32, but that would incur many other compilation errors. So the workaround for 32-bit ARM is to change the assignment to `val = math.MaxInt32`.

**Comment by benjaminbollen** [comment]
Please note that this will be overwritten as these files have moved in the restructuring of the consensus and pipe code.  The preferred solution will be to use `uint64` (or `uint32` if someone wants to argue for that as a max height), and replace the types consistently everywhere.

@zramsay thanks for linking this PR in a new issue

**Pull request merged**